### PR TITLE
Update client to FHIR v1.8

### PR DESCRIPTION
**Pull request by jawalonoski**
Update client to use v1.8 models and remove references to Conformance resource.

### CODE REVIEW

File: lib/fhir_client/client.rb [hunk 1]
```diff
@@ -134,8 +134,8 @@ def get_oauth2_metadata_from_conformance
       authorize_extension = 'authorize'
       token_extension = 'token'
       begin
-        conformance = conformance_statement
-        conformance.rest.each do |rest|
+        capabilities = capability_statement
```
Comment (wesrich) [inline lib/fhir_client/client.rb]: I'm not sure this variable is used anywhere, other than the line below. (Probably a bit nit-picky: but we probably could drop the temp variable.)

File: lib/fhir_client/client.rb [hunk 2]
```diff
@@ -170,10 +170,10 @@ def capability_statement(format = @default_format)
     # Method returns a conformance statement for the system queried.
     # @return
     def conformance_statement(format = @default_format)
-      if @cached_conformance.nil? || format != @default_format
+      if @cached_capability_statement.nil? || format != @default_format
```
Comment (wesrich) [inline lib/fhir_client/client.rb]: I think we have the `attr_accessor`, so we probably could drop the `@` here (and a couple other places), but not the end of the world either way.

File: lib/fhir_client/client.rb (hunk 1 above)
Comment (jawalonoski) [inline lib/fhir_client/client.rb]: Good point.

File: lib/fhir_client/client.rb (hunk 2 above)
Comment (jawalonoski) [inline lib/fhir_client/client.rb]: Huh... I sort of like the `@` -- makes it clear what we're dealing with here. _Shrugs_

**Pull request merged**